manifests/standalone.pp

```diff
@@ -0,0 +1,374 @@
+    }
+  }
+
+  if $chroot {
+    $_chroot = $chroot
+  }
+  elsif $facts['selinux_current_mode'] and $facts['selinux_current_mode'] == 'disabled' {
+    $_chroot = "/var/stunnel_${name}"
+  }
+  else {
+    $_chroot = undef
+  }
+
+  file { "/etc/stunnel/stunnel_${name}.conf":
+    ensure  => 'present',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0600',
+    content => template('stunnel/standalone_conf.erb'),
+    require => File['/etc/stunnel']
+  }
+
+  if $pki {
+    pki::copy { "stunnel_${name}":
+      source => $app_pki_external_source,
+      pki    => $pki,
+      notify =>  Service["stunnel_${name}"]
+    }
+  }
+
+  # NOTE: The pidfile directory is ensured by the service file(s)
+  if $_chroot {
+    file { $_chroot:
+      ensure => 'directory',
+      owner  => 'root',
+      group  => $setgid,
+      mode   => '0770',
+    }
+
+    # The following two entries are required to be able to properly resolve
+    # hosts using the _chroot directory.
+    file { "${_chroot}/etc":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0755',
+    }
+
+    file { "${_chroot}/etc/resolv.conf":
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      source => 'file:///etc/resolv.conf',
+    }
+
+    file { "${_chroot}/etc/nsswitch.conf":
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      source => 'file:///etc/nsswitch.conf',
+    }
+
+    file { "${_chroot}/etc/hosts":
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      source => 'file:///etc/hosts',
+    }
+
+    file { "${_chroot}/var":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644'
+    }
+
+    file { "${_chroot}/etc/pki":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => $setgid,
+      mode   => '0640'
+    }
+
+    file { "${_chroot}/etc/pki/cacerts":
+      source  => "file://${app_pki_dir}/cacerts",
+      group   => $setgid,
+      mode    => '0640',
+      recurse => true,
+      require =>  $pki ? { true =>  Pki::Copy["stunnel_${name}"], default =>  undef }
+    }
+  }
+
+  # The rules are pulled together from the accept_* and connect_*
+  # variables.
+  #
+  # This is only enabled if the system is a server.
+  if $firewall and !$client {
+    include '::iptables'
+
+    $_dport = [to_integer(split(to_string($accept),':')[-1])]
+
+    iptables::listen::tcp_stateful { "allow_stunnel_${name}":
+      trusted_nets => $trusted_nets,
+      dports       => $_dport
+    }
+  }
+
+  if !$client and $tcpwrappers {
+    include '::tcpwrappers'
+
+    tcpwrappers::allow { "allow_stunnel_${name}":
+      pattern => nets2ddq($trusted_nets)
+    }
+  }
+
+  if ($facts['os']['name'] in ['RedHat','CentOS']) {
```

Comment:
You can check to see if the `init_systems` fact contains systemd here as a more accurate and direct way to see what kind of service file to drop in